x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -170,7 +174,7 @@ public static Directory create(RepositoriesService repositories,
         if (SNAPSHOT_CACHE_ENABLED_SETTING.get(indexSettings.getSettings())) {
             final Path cacheDir = shardPath.getDataPath().resolve("snapshots").resolve(snapshotId.getUUID());
             directory = new CacheDirectory(directory, cache, cacheDir, snapshotId, indexId, shardPath.getShardId(),
-                currentTimeNanosSupplier);
+                indexSettings.getSettings(), currentTimeNanosSupplier);
```

Comment:
I don't think we need to pass all the index settings here, maybe just the black list?

Reply:
I was expecting other properties to be configured in the same way in the future, so found it easier to just pass the setting.